modules/xfem/include/materials/LevelSetBiMaterial.h

```diff
@@ -0,0 +1,59 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef LEVELSETBIMATERIAL_H
+#define LEVELSETBIMATERIAL_H
+
+#include "Material.h"
+
+// Forward Declarations
+class LevelSetBiMaterial;
+class XFEM;
+
+template <>
+InputParameters validParams<LevelSetBiMaterial>();
+
+/**
+ * Compute a material property for bi-materials defined by a level set function.
```

Comment:
Base class for switching between materials in a bi-material system where the interface is defined by a level set function.